cms/src/main/java/uk/nhs/digital/externalstorage/resource/S3NodeMetadata.java

```diff
@@ -0,0 +1,49 @@
+package uk.nhs.digital.externalstorage.resource;
+
+import uk.nhs.digital.externalstorage.ExternalStorageConstants;
+
+import javax.jcr.Node;
+import javax.jcr.RepositoryException;
+
+public class S3NodeMetadata {
+
+    public final long size;
+    public final String mimeType;
+    public final String fileName;
+    public final String url;
+
+    public S3NodeMetadata(Node node) {
+        size = getLongProperty(node, ExternalStorageConstants.PROPERTY_EXTERNALSTORAGE_SIZE);
+        mimeType = getStringProperty(node, ExternalStorageConstants.PROPERTY_EXTERNALSTORAGE_MIME_TYPE);
+        fileName = getStringProperty(node, ExternalStorageConstants.PROPERTY_EXTERNALSTORAGE_FILE_NAME);
+        url = getStringProperty(node, ExternalStorageConstants.PROPERTY_EXTERNALSTORAGE_PUBLIC_URL);
+    }
+
+    private String getStringProperty(Node node, String propertyName) {
+        try {
+            if (node.hasProperty(propertyName)) {
+                return node.getProperty(propertyName).getString();
+            } else {
+                return "";
+            }
+        } catch (RepositoryException ex) {
+            ex.printStackTrace();
```

Comment:
Should have proper error handling

Reply:
yes! Working on error handling today

Reply:
Can we have private fields with public getters rather than public fields?